nbdime/merging/generic.py

```diff
@@ -738,34 +738,74 @@ def _merge_concurrent_inserts(base, ldiff, rdiff, path, decisions, strategies=No
     into individual agreement/onesided/conflict decisions.
     """
     # Assume first ops are always inserts
+    assert 1 <= len(ldiff) <= 2
+    assert 1 <= len(rdiff) <= 2
     assert ldiff[0].op == DiffOp.ADDRANGE and rdiff[0].op == DiffOp.ADDRANGE
+    assert len(ldiff) == 1 or ldiff[1].op == DiffOp.REMOVERANGE
+    assert len(rdiff) == 1 or rdiff[1].op == DiffOp.REMOVERANGE
+
+
+    # FIXME: This function doesn't work out so well with new conflict handling,
+    # when an insert (e.g. [2,7] vs [3,7]) gets split into agreement on [7] and
+    # conflict on [2] vs [3], the ordering gets lost. I think this was always
+    # slightly ambiguous in the decision format, as the new inserts will get
+    # the same key and decisions are supposed to be possible to reorder (sort)
+    # without considering original ordering of decisions. To preserve the
+    # ordering, perhaps we can add relative local/remote indices to the decisions?
+    # We had this, where ordering made it work out correctly:
+    #   "conflicting insert [2] vs [3] at 1 (base index);
+    #    insert [7] at 1 (base index)"
+    # instead we now have this which messes up the ordering:
+    #   "insert [7] at 1 (base index);
+    #    conflicting insert [2] vs [3] at 1 (base index)"
+    # perhaps change to this:
+    #   "insert [7] at key=1 (base index) lkey=1 rkey=1;
+    #    conflicting insert [2] vs [3] at key=1 (base index) lkey=0 rkey=0"
+    # then decisions can be sorted on (key,lkey) or (key,rkey) depending on chosen side.
+    # This test covers the behaviour:
+    # py.test -k test_shallow_merge_lists_insert_conflicted -s -vv
```

Comment:
@vidartf something to discuss on Wednesday.

Note that this is related to how conflicts of inserts should be treated, and diff3 and git merge-file makes different choices:
```
$ more a.txt c.txt b.txt 
::::::::::::::
a.txt
::::::::::::::
1
4
3
::::::::::::::
c.txt
::::::::::::::
::::::::::::::
b.txt
::::::::::::::
1
2
3

$ diff3 -m a.txt c.txt b.txt 
<<<<<<< a.txt
1
4
3
||||||| c.txt
=======
1
2
3
>>>>>>> b.txt

$ git merge-file -p a.txt c.txt b.txt 
1
<<<<<<< a.txt
4
=======
2
>>>>>>> b.txt
3
```

Perhaps the easiest is to follow diff3 and ignore similarities between local and remote insertions.